Errors on `subctl verify` when running with Globalnet

<!-- Please use this template while reporting a bug and provide as much info as
possible. Not doing so may result in your bug not being addressed in a timely
manner. Thanks!

If the matter is security related, please disclose it privately to the
Submariner Owners: https://github.com/orgs/submariner-io/teams/submariner-core
-->


**What happened**:
Running `subctl verify` on 0.12.0-rc1 test day, the clusters are installed with Globalnet.
Got the following errors:
```bash
• Failure [295.955 seconds]
[discovery] Test Service Discovery Across Clusters
github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/service_discovery.go:42
  when a pod tries to resolve a service in a specific remote cluster by its cluster name
  github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/service_discovery.go:74
    should resolve the service on the specified cluster [It]
    github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/service_discovery.go:75

    Failed to verify if service IP is discoverable. expected execution result "" to contain "172.30.17.231"
    Unexpected error:
        <*errors.errorString | 0xc00018bea0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    occurred

    github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513
------------------------------
• Failure [285.577 seconds]
[discovery] Test Headless Service Discovery Across Clusters
github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/headless_services.go:37
  when a pod tries to resolve a headless service in a remote cluster
  github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/headless_services.go:40
    should resolve the backing pod IPs from the remote cluster [It]
    github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/headless_services.go:41

    Failed to  service IP verification. expected execution result "" to contain "242.1.255.252"
    Unexpected error:
        <*errors.errorString | 0xc00018bea0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    occurred

    github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513
------------------------------
• Failure [279.380 seconds]
[discovery] Test Stateful Sets Discovery Across Clusters
github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:34
  when a pod tries to resolve a podname from stateful set in a remote cluster
  github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:37
    should resolve the pod IP from the remote cluster [It]
    github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:38

    Failed to  service IP verification. expected execution result "" to contain "242.1.255.252"
    Unexpected error:
        <*errors.errorString | 0xc00018bea0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    occurred

    github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513
------------------------------
• Failure [280.104 seconds]
[discovery] Test Stateful Sets Discovery Across Clusters
github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:34
  when a pod tries to resolve a podname from stateful set in a local cluster
  github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:43
    should resolve the pod IP from the local cluster [It]
    github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:44

    Failed to  service IP verification. expected execution result "" to contain "242.0.255.252"
    Unexpected error:
        <*errors.errorString | 0xc00018bea0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    occurred

    github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513
------------------------------
• Failure [280.258 seconds]
[discovery] Test Stateful Sets Discovery Across Clusters
github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:34
  when the number of active pods backing a stateful set changes
  github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:49
    should only resolve the IPs from the active pods [It]
    github.com/submariner-io/lighthouse@v0.12.0-rc1/test/e2e/discovery/statefulsets.go:50

    Failed to  service IP verification. expected execution result "" to contain "242.1.255.250"
    Unexpected error:
        <*errors.errorString | 0xc00018bea0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    occurred

    github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513
------------------------------


Summarizing 6 Failures:

[Fail] [discovery] Test Service Discovery Across Clusters when a pod tries to resolve a service in a specific remote cluster by its cluster name [It] should resolve the service on the specified cluster 
github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513

[Fail] [discovery] Test Headless Service Discovery Across Clusters when a pod tries to resolve a headless service in a remote cluster [It] should resolve the backing pod IPs from the remote cluster 
github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513

[Fail] [discovery] Test Headless Service Discovery Across Clusters when a pod tries to resolve a headless service in a specific remote cluster by its cluster name [It] should resolve the backing pod IPs from the specified remote cluster 
github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513

[Fail] [discovery] Test Stateful Sets Discovery Across Clusters when a pod tries to resolve a podname from stateful set in a remote cluster [It] should resolve the pod IP from the remote cluster 
github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513

[Fail] [discovery] Test Stateful Sets Discovery Across Clusters when a pod tries to resolve a podname from stateful set in a local cluster [It] should resolve the pod IP from the local cluster 
github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513

[Fail] [discovery] Test Stateful Sets Discovery Across Clusters when the number of active pods backing a stateful set changes [It] should only resolve the IPs from the active pods 
github.com/submariner-io/shipyard@v0.12.0-rc1/test/e2e/framework/framework.go:513

Ran 24 of 41 Specs in 2129.496 seconds
FAIL! -- 18 Passed | 6 Failed | 0 Pending | 17 Skipped
[] E2E failed
```

Full log: [log.txt](https://github.com/submariner-io/lighthouse/files/8244410/log.txt)


**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Diagnose information (use `subctl diagnose all`): All passed
- Gather information (use `subctl gather`): [gather.tar.gz](https://github.com/submariner-io/lighthouse/files/8245204/gather.tar.gz)

- Cloud provider or hardware configuration: AWS, OCP 4.9
- Install tools:
- Others:

@nyechiel I had a close look at the logs and from Globalnet POV things are looking fine. There are some errors in Lighthouse tests and DNS resolution is failing because of which discovery tests are failing.

Apart from the observations I shared earlier, I also see some errors in LH when there is a request for DNS resolution and the entry was missing.

```
[ERROR] plugin/errors: 3 mkolesni-0.12-testday1.nginx-demo.e2e-tests-discovery-jnhsl.svc.clusterset.local. A: plugin/lighthouse: record not found
[ERROR] plugin/errors: 3 mkolesni-0.12-testday1.nginx-demo.e2e-tests-discovery-jnhsl.svc.clusterset.local. A: plugin/lighthouse: record not found
[ERROR] plugin/errors: 3 mkolesni-0.12-testday2.nginx-headless.e2e-tests-discovery-zfxpp.svc.clusterset.local. A: plugin/lighthouse: record not found
[ERROR] plugin/errors: 3 mkolesni-0.12-testday2.nginx-headless.e2e-tests-discovery-zfxpp.svc.clusterset.local. A: plugin/lighthouse: record not found
``` 

Since we no longer have the original setup, I'm trying to deploy AWS Clusters to check if the problem is transient or consistent.

@vthapar do you have any additional observations looking at Lighthouse logs?

Found the issue and confirmed from Coredns logs. Issue is only with headless/stateful set tests where we query a specific pod or pods in specific cluster.

```
[ERROR] plugin/errors: 3 mkolesni-0.12-testday2.nginx-headless.e2e-tests-discovery-zfxpp.svc.clusterset.local. A: plugin/lighthouse: record not found
[ERROR] plugin/errors: 3 mkolesni-0.12-testday2.nginx-headless.e2e-tests-discovery-zfxpp.svc.clusterset.local. A: plugin/lighthouse: record not found
```
Note that it is missing the podname for DNS query. It is because of `.` in cluster name. So it ends up interpreting `mkolesni-0` as podname and `12-testday2` as clustername. Clustername should be a valid DNS Label [alphanumeric and `-` only].

We should throw error to user when trying to deploy cluster with `.` in name. We can consider allowing it if servicediscovery is not enabled.

--- a/pkg/agent/controller/agent.go
+++ b/pkg/agent/controller/agent.go
@@ -38,6 +38,7 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	validations "k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/util/retry"
 	"k8s.io/klog"
@@ -60,6 +61,10 @@ var MaxExportStatusConditions = 10
 // nolint:gocritic // (hugeParam) This function modifies syncerConf so we don't want to pass by pointer.
 func New(spec *AgentSpecification, syncerConf broker.SyncerConfig, kubeClientSet kubernetes.Interface,
 	syncerMetricNames AgentConfig) (*Controller, error) {
+	if errs := validations.IsDNS1123Label(spec.ClusterID); len(errs) > 0 {
+		return nil, errors.Errorf("%s is not a valid ClusterID %v", spec.ClusterID, errs)
+	}
+
 	agentController := &Controller{
 		clusterID:        spec.ClusterID,
 		namespace:        spec.Namespace,

